src/NServiceBus.Persistence.DynamoDB/Config/DynamoDBPersistenceConfig.cs

```diff
@@ -28,15 +23,15 @@ public static PersistenceExtensions<DynamoDBPersistence> DynamoDBClient(this Per
         }
 
         /// <summary>
-        /// Sets the table name for the outbox and the saga storage
+        /// Uses the provided table configuration for both Saga and Outbox storage settings.
         /// </summary>
-        public static PersistenceExtensions<DynamoDBPersistence> TableName(this PersistenceExtensions<DynamoDBPersistence> persistenceExtensions, string tableName)
+        public static PersistenceExtensions<DynamoDBPersistence> UseSharedTable(this PersistenceExtensions<DynamoDBPersistence> persistenceExtensions, TableConfiguration sharedTableConfiguration)
         {
-            Guard.AgainstNullAndEmpty(nameof(tableName), tableName);
-
-            persistenceExtensions.Outbox().TableName = tableName;
-            persistenceExtensions.Sagas().TableName = tableName;
+            Guard.AgainstNull(nameof(persistenceExtensions), persistenceExtensions);
+            Guard.AgainstNull(nameof(sharedTableConfiguration), sharedTableConfiguration);
 
+            persistenceExtensions.Sagas().Table = sharedTableConfiguration;
```

Comment:
this does replace the reference, we could also use `sharedTableConfiguration with {}` since we're using records.